### PR TITLE
add stream elementnode 

### PR DESCRIPTION
**Pull request by houguohua**
modify wrong indent

### CODE REVIEW

**Comment by minggo** [comment]
Can not merge it pull request automatically. There are some conflicts.

**Comment by houguohua** [comment]
Please delete the old code in cocos2d/cocos2d-x.

File: cocos/3d/ElementNode.cpp [hunk 1]
```diff
@@ -0,0 +1,1102 @@
...
+
+	bool ElementNode::getElement(const char* name, Vector3* out) const
+	{
+	assert(out);
+
+	const char* valueString = getElement(name);
+	if (valueString)
+	{
+	float x, y, z;
+	int scanned;
```
Comment (minggo) [inline cocos/3d/ElementNode.cpp:939]: Wrong indent. Can fix it later.

File: cocos/3d/ElementNode.cpp [hunk 2]
```diff
@@ -0,0 +1,1102 @@
...
+		_flags.push_back(kv);
+	}
+
+	int ElementNode::getElementCount() const
+	{
+		return (int)_elements.size();
+	}
+
+
+}
```
Comment (minggo) [inline cocos/3d/ElementNode.cpp:1102]: It is better to add a space line at the end of a cpp file.

File: cocos/3d/ElementNode.h [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+	class Stream;
+
+	/**
+	* ElementNode is an interface for reading and writing element.
+	*/
+	class  ElementNode
+	{
+	public:
+
+		enum ElementType
```
Comment (minggo) [inline cocos/3d/ElementNode.h:23]: Can we use `enum class` here?

File: cocos/3d/Stream.cpp [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+		{
+			C3DFileStream* stream = new C3DFileStream(file);      
+
+			return stream;
+		}
+		return nullptr;
+	}
+
+
+	void C3DFileStream::close()
```
Comment (minggo) [inline cocos/3d/Stream.cpp:125]: Should we remove `C3D` prefix? I think it is base code, we can put it into `base` folder later.

File: cocos/3d/StringTool.cpp [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+            {
+                Result.push_back(std::string(pLastPos) );
+                pLastPos = pStart + i + strKey.size();
+                pLastPos = nullptr;
+            }
+            i += strKey.size() - 1;
+        }
+        else
+        {
+            if (!pLastPos) // 一个新字符串的开始
```
Comment (minggo) [inline cocos/3d/StringTool.cpp:53]: Chinese words.

File: cocos/3d/StringTool.cpp [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+    std::vector<std::string> Result;
+    const char *cpPos    = nullptr;
+    const char *pLastPos = nullptr;
+    char *pStart = new char[str.size() + 1];
+
+    strncpy(pStart, str.c_str(), str.size() + 1);
+
+    Result.clear();
+
+    pLastPos = nullptr;
```
Comment (minggo) [inline cocos/3d/StringTool.cpp:80]: Hungarian style.

File: cocos/3d/StringTool.cpp [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+{
+    b = str == "true";
+}
+
+
+
+
+
+
+
```
Comment (minggo) [inline cocos/3d/StringTool.cpp:176]: Unneeded spaces.

File: cocos/3d/ElementNode.cpp (hunk 1 above)
Comment (houguohua) [inline cocos/3d/ElementNode.cpp:939]: OK

File: cocos/3d/ElementNode.cpp (hunk 2 above)
Comment (houguohua) [inline cocos/3d/ElementNode.cpp:1102]: OK

**Pull request merged**